src/utils/BucketUtils.js

```diff
@@ -189,45 +189,65 @@ const filterDatasetsByDACs = (dacIds, datasets) => {
 /**
  * Generate the summary of algorithm results suitable for display in the UI
  *
- * Three potential cases:
+ * Four potential cases:
  *  1. No matches
  *  2. Exactly one match or N matches that are all the same - easy case
- *  3. N matches - not all the same - very confusing case
+ *  3. Abstain is true and match is false - decision is abstained
+ *  4. N matches - not all the same - very confusing case
  *
  * @private
  * @param bucket
  * @returns {{result: string, failureReasons: string[], id, createDate: undefined}|{result: (string|string), failureReasons: *, id: *, createDate: *}|{result: string, failureReasons: undefined, id, createDate: undefined}}
  */
 const calculateAlgorithmResultForBucket = (bucket) => {
-
-  // We actually DO NOT want to show system match results when the data use indicates
+  // V1 and V2: We actually DO NOT want to show system match results when the data use indicates
   // that a match should not be made. This happens for all "Other" cases.
+  const algorithmVersionV3 = bucket.matchResults.algorithmVersion === 'v3' ? true : false;
   const unmatchable = isOther(bucket.dataUse) || shouldAbstain(bucket.dataUse);
   // Check on all possible true/false values in the matches.
   // If all matches are the same, we can merge them into a single match object for display.
   // If they are not all the same, we have to punt this decision solely to the DAC.
-  const matchVals = unmatchable ? [] : flow(
+  // Check algorithm version: V3 does not need to be checked for 'unmatchable'
```

Comment:
The nested ternaries make this a little hard to follow. I think this could be expressed as:

```
const matchVals = (algorithmVersionV3 || !unmatchable) ? flow(
      map(m => m.match),
      uniq
    )(bucket.matchResults)) : [];
```

Reply:
Good point! Thanks for the suggestion